projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts

```diff
@@ -38,64 +41,51 @@ type ApiErrorMessages =
   templateUrl: './validation-messages.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ValidationMessagesComponent
-  implements OnInit, OnDestroy, DoCheck, AfterContentInit
-{
-  materialErrorMatcher = false;
-  shownErrors: string[] = [];
-
+export class ValidationMessagesComponent implements OnInit, AfterContentInit {
   @Input() errorsMessages: ValidationMessagesConfig = {};
   @Input() control!: FormControl;
   @Input() controlName!: string;
-
-  showServerErrors = false;
-  parsedApiErrorMessages: string[] = [];
-  valueChanges: Subscription | null = null;
-  private ngUnsubscribe: Subject<void> = new Subject<void>();
-
-  constructor(
-    @Host() protected host: MatFormField,
-    private cd: ChangeDetectorRef,
-    private validationMessagesService: ValidationMessagesService,
-    private controlContainer: ControlContainer
-  ) {
-    this.unsubscribeAndClearValueChanges =
-      this.unsubscribeAndClearValueChanges.bind(this);
-    this.materialErrorMatcher = validationMessagesService.materialErrorMatcher;
-  }
-
-  private _multiple = false;
-
-  get multiple(): boolean {
-    return this._multiple;
-  }
-
   @Input()
   set multiple(multiple: boolean) {
-    this._multiple = multiple;
-    this.updateErrorMessages();
+    this._multiple.set(multiple);
   }
-
-  private _apiErrorMessages: ApiErrorMessages = null;
-
-  get apiErrorMessages(): ApiErrorMessages {
-    return this._apiErrorMessages;
-  }
-
   @Input()
   set apiErrorMessages(apiErrorMessages: ApiErrorMessages) {
-    this.unsubscribeAndClearValueChanges();
-    this._apiErrorMessages = apiErrorMessages;
-    this.parseApiErrorMessages(this._apiErrorMessages);
-    this.showServerErrors = true;
+    this._apiErrorMessages.set(apiErrorMessages);
+  }
+
+  materialErrorMatcher = false;
+  showServerErrors = false;
+  shownErrors = computed(() => {
+    const control = this._formValueChange();
+    if (control && control.invalid && control.dirty) {
+      return this._defineErrorsToBeShown();
+    }
+    return [];
+  });
 
-    if (this.control && apiErrorMessages) {
+  shownApiErrors = computed(() => {
+    const apiErrors = this._apiErrorMessages();
+    this.showServerErrors = true;
+    if (this.control && apiErrors) {
       this.control.setErrors({
-        server: apiErrorMessages,
+        server: apiErrors,
       });
-
-      this.observeInputValueChanges();
     }
+    return this._parseApiErrorMessages(apiErrors);
+  });
+  private _formValueChange!: Signal<undefined | FormControl>;
+  private _multiple = signal(false);
+  private _apiErrorMessages = signal<ApiErrorMessages>(null);
+
+  constructor(
+    @Host() protected host: MatFormField,
+    private validationMessagesService: ValidationMessagesService,
+    private controlContainer: ControlContainer,
+    private destroyRef: DestroyRef,
+    private injector: Injector
```

Comment:
if you're using `_` for private fields, you should use them here as well to be consistent